python/taichi/idle_hacker.py

```diff
@@ -0,0 +1,54 @@
+# A dirty hack injector for python/taichi/lang/shell.py
+
+our_code = "__import__('taichi.idle_hacker').idle_hacker.idleipc(source)"
+
+
+def idleipc(source):
+    import os
+    import taichi as ti
+    taichi_dir = os.path.dirname(os.path.abspath(ti.__file__))
+    filename = os.path.join(taichi_dir, '.tidle_' + str(os.getpid()))
+    with open(filename, 'a') as f:
+        f.write('\n===\n' + source)
```

Comment:
We have to use this dirty method since idle and taichi is running in separate process... any idea?

Reply:
or we can directly wrap sockets and pass arbitrary python objects using multiprocessing? Correct me if I'm wrong, I feel like I'm missing sth here lol

Reply:
> sockets

In fact, the IDLE already using socket as IPC, and that could be an ultimate solution, however:
1. Not sure if it's invarient over different versions.
2. I'm not super familiar about sockets :(

So I'd rather write a portable file IPC instead, WDYT?

> ZeroMQ

No, Taichi is already a highly dedicated package, having too heavy dependencies could easily ban it from everyday people, sorry.

Reply:
Ahh I see. Then I think this solution's good enough. The one that works is always the best one.